docs/source/index.rst

```diff
@@ -49,9 +56,129 @@ a dialog into sub-dialogs and process them separately.
+
+- **SQL:** Provides a SQL-based version of the :py:class:`.DBContextStorage` class.
+   It allows the user to choose the backend option of his liking from MySQL, PostgreSQL, or SQLite.
+
+- **Mongo:** Provides a MongoDB-based version of the :py:class:`.DBContextStorage` class.
+
+- **JSON:** Provides a json-based version of the :py:class:`.DBContextStorage` class.
+
+- **Pickle:** Provides a pickle-based version of the :py:class:`.DBContextStorage` class.
+
+- **Database:** This module contains the Database class which is used to store and retrieve context data.
+
+- **Shelve:** Provides a shelve-based version of the :py:class:`.DBContextStorage` class.
+
+- **Yandex DB:** Provides a version of the :py:class:`.DBContextStorage` class that is specifically designed
+   to work with Yandex DataBase.
+
+
+Messenger Interfaces
+~~~~~~~~~~~~~~~~~~~~
+
+- **Message Interfaces:** This module contains several basic classes of message interfaces.
+   These classes provide a standardized way of interacting with different messaging services,
+   allowing the application to work with multiple messaging platforms seamlessly.
+
+- **dff.messengers.telegram package:** This package contains classes and functions specific
+   to the Telegram messenger service. It provides an interface for the application to interact with Telegram,
+   allowing it to send and receive messages, handle callbacks, and perform other actions.
+
+- **Types:** This module contains special types that are used for the messenger interface to client interaction.
+   These types are used to define the format of messages and other data that is exchanged between the
+   application and the messaging service.
+
+
+Pipeline
+~~~~~~~~
+
+- **Conditions:** This module contains a set of functions that are used to determine the possibility of transition
+   from one node of the dialog graph to another. Condition functions
+   are used to check the current state of the dialog and make decisions about how to proceed with the conversation.
+
+- **Service Group:** This module contains the :py:class:`.ServiceGroup` class. This class represents a group
+   of services that can be executed together in a specific order.
+
+- **Utils:** This module contains several utility functions that are used to perform various
+   tasks such as data processing, error handling, and debugging.
+
+- **Component:** This module contains the :py:class:`.PipelineComponent` class.
+   This class represents an individual component in a pipeline.
+
+- **Extra Handler:** This module contains an extra handlers that can be used to handle additional
+   functionality or features that are not included in the standard pipeline classes.
+
+- **Pipeline:** This module contains the :py:class:`.Pipeline` class. This class represents the main pipeline of
+   the `DFF` and is responsible for managing the execution of services.
+
+- **Service:** This module contains the :py:class:`.Service` class.
```

Comment:
is not clear

Reply:
**Service:** This module contains the :py:class:`.Service` class, which is used to represent a single service within the pipeline.